plugins/modules/nxos_ospfv2.py

```diff
@@ -0,0 +1,1610 @@
+                          initial_spf_delay:
+                            description:
+                              - Initial SPF schedule delay in milliseconds.
+                            type: int
+                          min_hold_time:
+                            description:
+                              - Minimum hold time between SPF calculations.
+                            type: int
+                          max_wait_time:
+                            description:
+                              - Maximum wait time between SPF calculations.
+                            type: int
+              vrf:
+                description:
+                  - Name/Identifier of the VRF.
+                type: str
+                required: True
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    - gathered
+    - parsed
+    - rendered
+    default: merged
+"""
+EXAMPLES = """
+# Using merged
+
+# Before state:
+# -------------
+# nxos-9k-rdo# sh running-config | section "^router ospf .*"
+# nxos-9k-rdo#
+
+- name: Merge the provided configuration with the exisiting running configuration
+  cisco.nxos.nxos_ospfv2:
+    config:
+      processes:
+        - process_id: 100
+          router_id: 203.0.113.20
+        - process_id: 102
+          router_id: 198.51.100.1
+          areas:
+            - area_id: 0.0.0.100
+              filter_list:
+                - route_map: rmap_1
+                  direction: in
+                - route_map: rmap_2
+                  direction: out
+              ranges:
+                - prefix: 198.51.100.64/27
+                  not_advertise: True
+                - prefix: 198.51.100.96/27
+                  cost: 120
+            - area_id: 0.0.0.101
+              authentication:
+                message_digest: True
+          redistribute:
+            - protocol: eigrp
+              id: 120
+              route_map: rmap_1
+            - protocol: direct
+              route_map: ospf102-direct-connect
+          vrfs:
+            - vrf: zone1
+              router_id: 198.51.100.129
+              redistribute:
+                - protocol: static
+                  route_map: zone1-static-connect
+              summary_address:
+                - prefix: 198.51.100.128/27
+                  tag: 121
+                - prefix: 198.51.100.160/27
+              areas:
+                - area_id: 0.0.0.102
+                  nssa:
+                    default_information_originate: True
+                    no_summary: True
+                - area_id: 0.0.0.103
+                  nssa:
+                    no_summary: True
+                    translate:
+                      type7:
+                        always: True
+            - vrf: zone2
+              auto_cost:
+                reference_bandwidth: 45
+                unit: Gbps
+    state: merged
+
```

Comment:
can u also include command fired, as it gives more visibility to the users for all of the valid states

Reply:
Ack.